community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/Instruction.scala

```diff
@@ -28,16 +30,20 @@ trait Instruction {
   //generate class level members
   def fields(): String
 
-  // Initialises necessary data-structures. Is inserted at the top of the generated method
+  //Initialises necessary data-structures. Is inserted at the top of the generated method
   def generateInit(): String
 
   def methods: Seq[Method] = {
     (allLeafs :+ this).flatMap(_._method)
   }
 
+  def exceptions: Set[ExceptionCodeGen] = allLeafs.flatMap(_._exceptions()).toSet
+
   // Generates import list for class
   protected def _importedClasses(): Set[String] = Set.empty
 
+  protected def _exceptions(): Set[ExceptionCodeGen] = Set.empty
```

Comment:
We could make this abstract to clarify the difference between it and the non-underscore variant, but that would on the other hand mean duplicated `Set.empty` implementations in subclasses... I'm torn on this one.